### DISCUSSION TITLE
User permission for  Constraints

### DISCUSSION THREAD

**Discussion by alemsas** [netbox-community/netbox#14352] [General]
I want to write a JSON expression for DCIM>Devices that user just can update the status of each devices!
anyone can help me??

**Comment by candlerb** [comment]
If by "JSON" you mean [Custom Validators](https://docs.netbox.dev/en/stable/customization/custom-validation/#custom-validation-logic) then AFAIK they can't see the user making the change, nor the before-and-after attribute values. [Permissions](https://docs.netbox.dev/en/stable/administration/permissions/) only work at the level of entire objects, not individual attributes.

Your best bet, I think, would be to configure permissions so that the user cannot modify the device at all (i.e. withhold the "change" and "delete" permissions), and then write a [Custom Script](https://docs.netbox.dev/en/stable/customization/custom-scripts/) (which the user has permission to run) which lets them select a device and status to change.

You may want to add a [Custom Link](https://docs.netbox.dev/en/stable/customization/custom-links/) (button) on the Device page, to allow them more easily to navigate to the script to update the device status.

**Reply by alemsas** [reply]
I will try your solutions!

**Reply by alemsas** [reply]
Is there any example for this?!I use netbox (v3.6.5)!

**Reply by candlerb** [reply]
Yes, lots of examples of custom scripts at https://github.com/netbox-community/customizations

**Reply by alemsas** [reply]
thanks I wrote and ran my scripts!
but when I want to share these scripts with other users the permissions on object types doesnt work and the user get 
No Scripts Found!
here is 
Object Types

    extras | script
    extras | script module

and I get it view and change action!

**Reply by candlerb** [reply]
See https://docs.netbox.dev/en/stable/customization/custom-scripts/#running-custom-scripts

**Reply by alemsas** [reply]
again thanks!
I do above
![image](https://github.com/netbox-community/netbox/assets/67168373/7acc47a5-2f53-4ca0-b2da-aa5c388dfc62)
but when I want to run with the traget user i get the error that you can see it in the right and bottem of my screenshot!
also I do this for user Extras > Script, Extras > ScriptModule, and Core > ManagedFile!

**Reply by alemsas** [reply]
the new role has been loaded but the device id not!

**Reply by candlerb** [reply]
I suspect the user still needs View permission on devices, so that the drop-down field can be populated.

BTW, to get rid of the scheduler settings (which you almost never want), add the following to your class:

```
class ChangeDeviceRole(Script):
    class Meta:
        scheduling_enabled = False
```

**Comment by alemsas** [comment]
It has been solved with above works!
thanks @candlerb
